ggml/src/gguf.cpp

```diff
@@ -1321,14 +1341,59 @@ void gguf_write_to_buf(const struct gguf_context * ctx, std::vector<int8_t> & bu
         return;
     }
 
-    const size_t offset_data = gw.buf.size();
+    const size_t offset_data = gw.written_bytes;
 
     // write tensor data
     for (int64_t i = 0; i < n_tensors; ++i) {
         gw.write_tensor_data(ctx->info[i], offset_data, ctx->alignment);
     }
 }
 
+// file based writer
+struct gguf_writer_file final : public gguf_writer_base {
```

Comment:
Please make it so that `gguf_writer_file` stops trying to write to the file once `ok` is false. Also please move it upwards so that the writer structs are all next to each other.

Reply:
> Please make it so that gguf_writer_file stops trying to write to the file once ok is false.

Not a perfect solution of breaking out of writing the file, but this turns writes effectively into nops.

> Also please move it upwards so that the writer structs are all next to each other.

done.